core/server/common/src/main/java/alluxio/Registry.java

```diff
@@ -36,47 +36,49 @@
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * Class for registering individual masters that make up the Alluxio master service. The intended
- * use of this class is for the Alluxio master service to register the individual masters using an
+ * Class for registering individual servers that run within an Alluxio process. The intended
+ * use of this class is for the Alluxio process to register the individual servers using an
  * instance of this class. The registry is passed as an argument to constructors of individual
- * masters who are expected to add themselves to the registry.
+ * servers who are expected to add themselves to the registry.
  *
  * The reason for using an instance as opposed to a static class is to enable dependency
  * injection in tests.
+ *
+ * @param <T> the type of the server
  */
 @ThreadSafe
-public final class MasterRegistry {
+public final class Registry<T extends Server> {
   private static final int TIMEOUT_SECONDS = 1;
   private static final int RETRY_COUNT = 5;
 
-  private final Map<Class<?>, Master> mRegistry = new HashMap<>();
+  private final Map<Class<?>, T> mRegistry = new HashMap<>();
```

Comment:
`Class<? extends T>` for more type safety. All classes in this map should extend `T`